src/main/java/katas/Kata10.java

```diff
@@ -55,9 +57,21 @@
         List<Map> lists = DataUtil.getLists();
         List<Map> videos = DataUtil.getVideos();
 
-        return ImmutableList.of(ImmutableMap.of("name", "someName", "videos", ImmutableList.of(
-                ImmutableMap.of("id", 5, "title", "The Chamber"),
-                ImmutableMap.of("id", 3, "title", "Fracture")
-        )));
+        //return ImmutableList.of(ImmutableMap.of("name", "someName", "videos", ImmutableList.of(
+        //ImmutableMap.of("id", 5, "title", "The Chamber"),
+        //ImmutableMap.of("id", 3, "title", "Fracture")
+        //)));
+        return DataUtil.getLists()
+                .stream()
+                .map(listMap -> ImmutableMap.of(
+                        "name", listMap.get("name"),
+                        "videos", DataUtil.getVideos()
+                                .stream()
+                                .filter(filterMap -> filterMap.get("listId").equals(listMap.get("id")))
+                                .map(videoMap -> ImmutableMap.of(
+                                        "id", videoMap.get("id"),
+                                        "title", videoMap.get("title")))
+                                .collect(Collectors.toList())))
```

Comment:
[mf] pls extract this huge block of code into separate private methods for better readability